java/src/client/model/Populator.java

```diff
@@ -31,15 +32,24 @@
 public class Populator extends Observable implements iPopulator {
 
     private ClientModel model;
+    private static Populator instance;
+    private ModelFacade facade;
+    
+    public static Populator getInstance() {
+        if (instance == null) {
+            instance = new Populator();
+        }
+        return instance;
+    }
 
-    public Populator() {
-        model = new ClientModel();
+    private Populator() {
+        model = Data.getCurentModelInstance();
+        facade = new ModelFacade();
```

Comment:
Just a thought, initialize Model Facade outside of constructor, and pass in the model (replaces old model)